public/app/plugins/datasource/loki/LanguageProvider.test.ts

```diff
@@ -268,6 +268,23 @@ describe('Language completion provider', () => {
         end: expect.any(Number),
       });
     });
+
+    it('should use a single promise to resolve values', async () => {
+      const datasource = setup({ testkey: ['label1_val1', 'label1_val2'], label2: [] });
+      const provider = await getLanguageProvider(datasource);
+      const requestSpy = jest.spyOn(provider, 'request');
+      const promise1 = provider.fetchLabelValues('testkey');
+      const promise2 = provider.fetchLabelValues('testkey');
+      const promise3 = provider.fetchLabelValues('testkeyNOPE');
+      expect(requestSpy).toHaveBeenCalledTimes(2);
```

Comment:
Not really, because I'm asserting 1 API call and 2 cached promises returned. In any case, I added an extra `      expect(requestSpy).toHaveBeenCalledTimes(2);` for better resilience to changes. Thanks for the suggestion.